indico/modules/events/registration/apple_pass.py

```diff
@@ -0,0 +1,133 @@
+            .sign(serialization.Encoding.DER, options)
+        )
+
+
+class ApplePassManager:
+    """Apple Pass ticketing manager.
+
+    This class handles the integration with the Apple Pass ticketing API.
+    """
+
+    def __init__(self, event):
+        self.event = event
+        self.settings = self.event.category.effective_apple_pass_config if not self.event.is_unlisted else None
+        self.cert = None
+        self.static_path = os.path.join(current_app.root_path, 'web', 'static')
+
+    @property
+    def configured(self):
+        return config.ENABLE_APPLE_PASS and self.settings is not None
+
+    def build_ticket_object(self, registration):
+        from indico.modules.categories.controllers.util import make_format_event_date_func
+        ticket = EventTicket()
+        ticket.addPrimaryField('event-title', self.event.title, 'Event')
+        event_date = make_format_event_date_func(self.event.category)(self.event)
+        ticket.addSecondaryField('event-date', event_date, 'Date')
+        if self.event.address and self.event.has_location_info:
+            venue_name = (f'{self.event.room_name} ({self.event.venue_name})'
+                          if self.event.room_name and self.event.venue_name
+                          else (self.event.venue_name or self.event.room_name))
+            ticket.addSecondaryField('event-venue', venue_name, 'Venue')
+        ticket.addAuxiliaryField('registration-name', registration.full_name, 'Name')
+        ticket.addAuxiliaryField('registration-email', registration.email, 'Email')
+        ticket.addBackField('back-registration-name', registration.full_name, 'Name')
+        ticket.addBackField('back-event-venue', venue_name, 'Venue')
+        ticket.addBackField('back-ticket-number', f'#{registration.friendly_id}', 'Ticket number')
+        ticket.addBackField('back-event-date', event_date, 'Date')
+        ticket.addBackField('back-registration-email', registration.email, 'Email')
+        ticket.addBackField('back-event-url', f'<a href="{self.event.external_url}">Indico event page</a>', 'Link')
+        reg_url = url_for('event_registration.display_regform', registration.locator.uuid, _external=True)
+        ticket.addBackField('back-registration-details', f'<a href="{reg_url}">Registration details</a>', 'Link')
+
+        signals.event.registration.apple_pass_ticket_object.send(self.event, obj=ticket)
+        return ticket
+
+    def build_pass_object(self, registration):
+        from indico.modules.events.registration.util import get_persons, get_ticket_qr_code_data
+
+        # Extract Certificate details from certificate.pem itself
+        self.cert = x509.load_pem_x509_certificate(self.settings['apple_pass_certificate'].encode('UTF-8'))
+        cert_details = {d.split('=')[0]: d.split('=')[1] for d in self.cert.subject.rfc4514_string().split(',')}
+        passfile = IndicoPass(self.build_ticket_object(registration), passTypeIdentifier=cert_details['UID'],
```

Comment:
I'm confused with this variable name. This is not the actual pass file but a pass object. Why not simply calling it `pass`?

```suggestion
        pass = IndicoPass(self.build_ticket_object(registration), passTypeIdentifier=cert_details['UID'],
```

Reply:
`pass` is a reserved keyword in python :p

Reply:
🙈